### PR TITLE
add open ocall function and resolve comments

### PR DESCRIPTION
**Pull request by johnvenn**
fix ssize issue in write / read

### CODE REVIEW

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp [hunk 1]
```diff
@@ -99,19 +99,40 @@ size_t sgxssl_read(int fd, void *buf, size_t count)
...
 
 	return retval;
 }
 
-// TODO
+int sgxssl_open(const char *filename, int flags)
+{
+	FSTART;
+
+    if (filename == NULL) return -1;
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_tsgxssl/tunistd.cpp:118]: Please notice the indent.

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp [hunk 2]
```diff
@@ -99,19 +99,40 @@ size_t sgxssl_read(int fd, void *buf, size_t count)
...
+	// In addition, the function is used by b_sock2.c as closesocket function.
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_tsgxssl/tunistd.cpp:119]: Suggest to remove the unnecessary comments.

File: Linux/sgx/libsgx_usgxssl/uunistd.cpp [hunk 3]
```diff
@@ -52,4 +53,10 @@ extern "C" {
 		return ret;
 	}
 
+    int u_sgxssl_open(const char *filename, int flags)
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_usgxssl/uunistd.cpp:56]: Please notice the indent.

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp [hunk 4]
```diff
@@ -99,19 +99,40 @@ size_t sgxssl_read(int fd, void *buf, size_t count)
...
+	// It is unreachable under the assumption that TLS support is not required.
+	// Otherwise should be implemented as OCALL.
+	int retval = 0;
+	sgx_status_t ret =
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_tsgxssl/tunistd.cpp:123]: Suggest to write the code of line 123 and 124 in one line.

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp [hunk 5]
```diff
@@ -71,19 +71,19 @@ size_t sgxssl_write (int fd, const void *buf, size_t n)
 	// It is unreachable under the assumption that TLS support is not required.
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_tsgxssl/tunistd.cpp:71]: Not sure if this comment is still necessary. I think the API can be used not only in TLS.

Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tunistd.cpp:71]: the comment is not unnecessary and removed. standard lib C APIs for file open/close/read/write.

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp (hunk 4 above)
Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tunistd.cpp]: updated

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp (hunk 1 above)
Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tunistd.cpp]: updated.

**Review by ShundaZhang — APPROVED** [review]

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp (hunk 2 above)
Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tunistd.cpp]: removed.

File: Linux/sgx/libsgx_usgxssl/uunistd.cpp (hunk 3 above)
Comment (johnvenn) [reply Linux/sgx/libsgx_usgxssl/uunistd.cpp]: updated.

File: Linux/package/include/sgx_tsgxssl.edl [hunk 6]
```diff
@@ -32,14 +32,18 @@
...
-    
+   
     untrusted {
         void u_sgxssl_ftime([out, size=timeb_len] void * timeptr, uint32_t timeb_len);
-        size_t u_sgxssl_write (int fd, [in,size=n]const void *buf, size_t n);
-        size_t u_sgxssl_read(int fd, [out, size=count]void *buf, size_t count);
+        ssize_t u_sgxssl_write (int fd, [in,size=n]const void *buf, size_t n);
+        ssize_t u_sgxssl_read(int fd, [out, size=count]void *buf, size_t count);
         int u_sgxssl_close(int fd);
+        int u_sgxssl_open([user_check]const char* fname, int flags);
```
Comment (llly) [inline Linux/package/include/sgx_tsgxssl.edl:43]: Not for this PR.
When these file related untrusted functions fails, `errno` represents error code.  `On error, -1 is returned, and errno is set appropriately.`
Can you add `propagate_errno` EDL key word to in order to pass `errno` to enclave?

**Review by llly — APPROVED** [review]

**Pull request merged**